test/node/medtronic/testMedtronicSimulator.js

```diff
@@ -0,0 +1,504 @@
+        .with_status('suspended')
+        .with_duration(600000)
+        .with_reason({resumed: 'automatic'})
+        .done();
+
+      var suspendedBasal = builder.makeSuspendBasal()
+        .with_time('2014-09-25T02:00:00.000Z')
+        .with_deviceTime('2014-09-25T02:00:00')
+        .with_timezoneOffset(0)
+        .with_conversionOffset(0);
+
+      var basal2 = builder.makeScheduledBasal()
+        .with_time('2014-09-25T03:00:00.000Z')
+        .with_deviceTime('2014-09-25T03:00:00')
+        .with_timezoneOffset(0)
+        .with_conversionOffset(0)
+        .with_scheduleName('Alice')
+        .with_rate(2);
+
+      var expectedSuspendedBasal = _.cloneDeep(suspendedBasal);
+      var suppressed = {
+        type: 'basal',
+        deliveryType: 'scheduled',
+        rate: 1
+      };
+      expectedSuspendedBasal.duration = 3600000;
+      expectedSuspendedBasal.set('suppressed', suppressed);
+
+      simulator.basal(basal);
+      simulator.suspendResume(suspendResume);
+      simulator.basal(suspendedBasal);
+      simulator.basal(basal2);
+
+      expect(simulator.getEvents()).deep.equals([
+        basal.done(),
+        suspendResume,
+        expectedSuspendedBasal.done(),
+      ]);
+
+    });
+
+    describe('temp basal', function() {
+
+      var basal1 = null, tempBasal = null, basal2 = null;
+      beforeEach( function() {
+        basal1 = builder.makeScheduledBasal()
+          .with_time('2014-09-25T17:05:00.000Z')
+          .with_deviceTime('2014-09-25T18:05:00')
+          .with_timezoneOffset(60)
+          .with_conversionOffset(0)
+          .with_rate(1.3)
+          .with_duration(2000000)
+          .set('index',0);
+        tempBasal = builder.makeTempBasal()
+          .with_time('2014-09-25T17:10:00.000Z')
+          .with_deviceTime('2014-09-25T18:10:00')
+          .with_timezoneOffset(60)
+          .with_conversionOffset(0)
+          .with_duration(1800000)
+          .with_rate(1.0)
+          .set('index',1);
+        basal2 = builder.makeScheduledBasal()
+          .with_time('2014-09-25T17:40:00.000Z')
+          .with_deviceTime('2014-09-25T18:40:00')
+          .with_timezoneOffset(60)
+          .with_conversionOffset(0)
+          .with_rate(2)
+          .with_duration(1800000)
+          .set('index',2);
+      });
+
+      it('adds supressed info', function() {
+
+        var expectedTempBasal = tempBasal.set('suppressed',{
+          type: 'basal',
+          deliveryType: 'scheduled',
+          rate: 1.3
+        }).done();
+
+        delete basal1.index;
+        delete basal1.jsDate;
+        delete expectedTempBasal.index;
+        delete expectedTempBasal.jsDate;
+
+        simulator.basal(basal1);
+        simulator.basal(tempBasal);
+        simulator.basal(basal2);
+        expect(simulator.getEvents()).deep.equals([
+          basal1.done(),
+          expectedTempBasal
+        ]);
+      });
+
+      it('checks for schedule change', function() {
```

Comment:
Add another test case here of a temp basal crossing midnight, possibly with and without the rate actually *changing* at midnight.